### PR TITLE
prevent mitm proxy auth to be escaped prior to base64 encoding it

### PR DESCRIPTION
**Pull request by GlenDC**
<https://cs.opensource.google/go/go/+/refs/tags/go1.20:src/net/url/url.go;l=420-431> shows that the auth info (both username and password) gets escaped, which is undesirable as their escape is also escaping legit chars such as '!'.

This commits disables the escaping completly. We can also define our own escaping if you think this is useful, but I doubt that for our use case we really need to worry about this and can expect the user to pass in valid credentials. Escaping would either way not help here as you'll just replace one error (invalid chars) into another error (not authorized)

### CODE REVIEW

**Comment by blakebyrnes** [comment]
This isn't building in go 1.17. How did you test?

**Comment by GlenDC** [comment]
> This isn't building in go 1.17. How did you test?

not 😰 
I guess I should, haha, let me fix the compiler errors tonight,
for sure will be something minor, did this purely from the editor.

My appologies

**Comment by GlenDC** [comment]
Should be okay now @blakebyrnes  :)

File: agent/mitm-socket/go/generate_cert.go [hunk 1]
```diff
@@ -80,11 +80,11 @@ func NewAuthority() (*x509.Certificate, *rsa.PrivateKey, error) {
 	certFromDisk, err := readCertFromDisk(caFile)
 
 	if err != nil && !errors.Is(err, os.ErrNotExist) {
-		log.Printf("Error reading cert from disk", caFile, err)
+		log.Print("Error reading cert from disk", caFile, err)
```
Comment (blakebyrnes) [inline agent/mitm-socket/go/generate_cert.go:83]: Did gofmt automatically change these? Or is there a reason to change them to Print?

Comment (GlenDC) [reply agent/mitm-socket/go/generate_cert.go:83]: You should get the warning with std go tooling, even just your compiler.
Anyway, question is more, why Printf in the first place, you're not formatting any args :P

**Comment by GlenDC** [comment]
Playground example: https://go.dev/play/p/q2c1YXByUd9

File: agent/mitm-socket/go/generate_cert.go (hunk 1 above)
Comment (blakebyrnes) [reply agent/mitm-socket/go/generate_cert.go:83]: Ah, good point.. ha

**Pull request merged**